evm/src/SolverAccount.sol

```diff
@@ -0,0 +1,235 @@
+        DOMAIN_SEPARATOR = IIntentGatewayV2(intentGatewayV2).DOMAIN_SEPARATOR();
+    }
+
+    /**
+     * @notice Validates a user operation before execution
+     * @dev Implements ERC-4337 validation logic with two modes:
+     *
+     *      Mode 1 - Standard validation (65-byte signature):
+     *      - Delegates to parent Account implementation
+     *      - Owner signs the userOpHash
+     *
+     *      Mode 2 - Intent solver selection (85-byte signature):
+     *      - Signature format: abi.encodePacked(sessionKey, ownerSignature)
+     *      - CallData format: IntentGatewayV2.select(SelectOptions)
+     *      - Validates that:
+     *        1. Calldata length and function selector are correct for select()
+     *        2. Solver in calldata matches address(this)
+     *        3. Nonce from EntryPoint (using first 192 bits of commitment as key)
+     *        4. Owner signed EIP-712 message: SelectSolver(commitment, sessionKey, nonce)
+     *      - NOTE: This function only validates. The bundler will execute the callData
+     *        (IntentGateway.select) via this contract's fallback function after validation.
+     *      - Replay protection: Signature includes nonce from EntryPoint
+     *
+     * @param op The packed user operation containing calldata, signature, and other fields
+     * @param userOpHash The hash of the user operation (with EntryPoint and chain ID)
+     * @param missingAccountFunds The amount of funds missing in the account to pay for gas
+     * @return validationData A packed value indicating validation result and time range
+     *         - 0 indicates successful validation
+     *         - 1 indicates signature validation failure
+     *         - Other values can encode time ranges for signature validity
+     */
+    function validateUserOp(PackedUserOperation calldata op, bytes32 userOpHash, uint256 missingAccountFunds)
+        public
+        override
+        onlyEntryPoint
+        returns (uint256)
+    {
+        if (op.signature.length == ECDSA_SIGNATURE_LENGTH) {
+            return super.validateUserOp(op, userOpHash, missingAccountFunds);
+        }
+
+        if (
+            op.callData.length != SELECT_CALLDATA_LENGTH || bytes4(op.callData[0:4]) != SELECT_SELECTOR
+                || op.signature.length != INTENT_SELECT_SIGNATURE_LENGTH
+        ) {
+            return ERC4337Utils.SIG_VALIDATION_FAILED;
+        }
+
+        // Decode SelectOptions from calldata (skip 4-byte selector)
+        SelectOptions memory options = abi.decode(op.callData[4:], (SelectOptions));
+        if (options.solver != address(this)) return ERC4337Utils.SIG_VALIDATION_FAILED;
+
+        address sessionKey = address(bytes20(op.signature[0:20]));
+        bytes calldata solverSignature = op.signature[20:];
+
+        // Get nonce from EntryPoint using first 192 bits of commitment as key
+        uint192 nonceKey = uint192(uint256(options.commitment) >> 64);
```

Comment:
Should we use a constant for the key just so that we don't bloat the chain's storage?